x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/results/TextEmbeddingResults.java

```diff
@@ -59,6 +61,40 @@ public TextEmbeddingResults(StreamInput in) throws IOException {
         );
     }
 
+    public static TextEmbeddingResults of(List<? extends InferenceResults> results) {
+        List<Embedding> embeddings = new ArrayList<>(results.size());
+        for (InferenceResults result : results) {
+            if (result instanceof org.elasticsearch.xpack.core.ml.inference.results.TextEmbeddingResults embeddingResult) {
+                embeddings.add(new Embedding(convertEmbeddingResultToEmbedding(embeddingResult)));
+            } else if (result instanceof org.elasticsearch.xpack.core.ml.inference.results.ErrorInferenceResults errorResult) {
+                if (errorResult.getException() instanceof ElasticsearchStatusException statusException) {
+                    throw statusException;
+                } else {
+                    throw new ElasticsearchStatusException(
+                        "Received error inference result: " + errorResult.getResultsField(),
+                        RestStatus.INTERNAL_SERVER_ERROR,
+                        errorResult.getException()
```

Comment:
Looking at this bit of code again, I am thinking it doesn't make a ton of sense. I think the intention of this function is to sort out the error results from the real results, but then I am not sure what we should do with the errors. Here I am just throwing the errors as an exception, but I am realizing that I am logging the resultsField here, which is sounds like a problem. 

Overall, I'm not sure what we should be doing with these error results when we get them. Is there a specific place we should be handling them?